impl/client/src/main/javascript/web/dojo/pentaho/common/FilterDialog.js

```diff
@@ -419,11 +419,13 @@ define(["dojo/_base/declare", "dijit/_WidgetBase", "dijit/_TemplatedMixin", "dij
 
   _unescapeAmpIfExist: function(str) {
     if (Array.isArray(str)) {
-      str.forEach(function(val) {
-        val.replace(/&amp;/g, '&');
+      str.forEach(function(val, index) {
+        if (typeof str === 'string' || str instanceof String) {
```

Comment:
Should it be: `if (typeof val === 'string' || val instanceof String)`
You should start testing your changes...

Reply:
Sorry, my bad...